nh_eobs/tests/report_nh_clinical_observation_report/test_process_report_dates.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+""" Module for TestProcessReportDates class. """
+from openerp.tests.common import TransactionCase
+
+from openerp.addons.nh_eobs.report.helpers import BaseReport
+
+
+class TestProcessReportDates(TransactionCase):
+    """
+    Test ProcessReportDates method of the
+    report.nh.clinical.observation_report model.
+    """
+    def setUp(self):
+        super(TestProcessReportDates, self).setUp()
+        self.report_model = self.env['report.nh.clinical.observation_report']
+        self.report_wizard_model = \
+            self.env['nh.clinical.observation_report_wizard']
+
+        self.data = self.report_wizard_model.create({
+            'start_time': '2017-03-05 15:00:00',
+            'end_time': '2017-05-24 15:00:00'
+        })
+        self.data_empty = self.report_wizard_model.create({
+            'start_time': False,
+            'end_time': False
+        })
+        self.spell_dictionary = {
+            'date_started': '2017-01-01 14:00:00',
+            'date_terminated': '2017-06-06 14:00:00'
+        }
+        self.spell_dictionary_no_date_terminated = {
+            'date_started': '2017-01-01 14:00:00',
+            'date_terminated': False
+        }
+        # Only interested in object having a value for 'time generated'.
+        self.base_report = BaseReport(None, None, None, '2017-07-06 10:00:00')
+
+    def call_test(self, data=None, spell=None, base_report=None):
```

Comment:
Could add doc string to explain what the different parameters means when calling the test